### DISCUSSION TITLE
So chatgpt is still limited to the 4096 token context length?

### DISCUSSION THREAD

**Discussion by aliencaocao** [acheong08/ChatGPT#649] [Q&A] [answered]
I saw the code here https://github.com/acheong08/ChatGPT/blob/main/src/revChatGPT/Official.py#L282 and it seems to me that this is exactly the same as gpt-3 in the sense that you are still limited to past 4096 tokens of chat history, since you have to prepend the chat history to the prompt. So what exactly has changed, besides using a different model perhaps fine-tuned for using chat history as context?
I am looking for a solution that does not have such context limit, but so far even chatgpt seems like this. If this is how chatgpt achieves contextual awareness, then I can just implement my own function to prepend chat history to the prompt and send to GPT-3. I am quite confused about why people prefer this than GPT-3.

I also took a look at the API requests on the webpage, and it is not obvious to me that they sent the entire conversation per request. So how did you come up with this implementation? Did you actually find evidence of openAI doing this to their prompts, or you are just trying it out and it happened to work?

**Comment by acheong08** [comment, marked as answer]
yes

**Reply by aliencaocao** [reply]
yes to the token length limit? Have you tried on the web interface/found any evidence or is this just based on how you assumed openAI implements stuff (since you implemented this way)? Because if this is true, then the whole hype about contextual awareness is quite perplexing, if it just forgets chat history more than 4000 tokens earlier.

**Reply by acheong08** [reply]
Contextual awareness is a lie. If you check the source code of our server https://github.com/acheong08/ChatGPT-Proxy-V2/ you can see that it uses the same model as the website requests.

**Reply by acheong08** [reply]
It's just buffered memory

**Reply by acheong08** [reply]
There is a small difference which is that the conversations endpoint puts it through a few layers of processing before passing it to completions, probably to compress it via another model

**Reply by aliencaocao** [reply]
that is very surprising to hear. Thanks for the answer

**Reply by tzekid** [reply]
Sorry to necro this discussion, but how do you guys compress the text for longer conversations? Just recursively ask chatgpt to summarise the discussion and feed that into a system or assistant message?

**Reply by acheong08** [reply]
Not implemented but based on some work by @waylaidwanderer, you can do that via embeddings for long term memory

**Reply by tzekid** [reply]
That's a good point of reference. Just discovered [OpenAI's Embeddings](https://platform.openai.com/docs/guides/embeddings/what-are-embeddings), might look for something on-device for my use case.
Thanks!
